### DISCUSSION TITLE
Notification when task is changed

### DISCUSSION THREAD

**Discussion by hpposch** [elsa-workflows/elsa-core#1557] [Q&A]
Hi all,
i was wondering if there is a way to get notified when there is a change from one task to another.
For example my workflow is waiting because of a timer task that runs for a long time.
So i do not want to poll the status every few seconds.
It would be great if there is something like an event that gets fired when this happens.
Is there anything like this?
Or are there other ways to do this?

Thanks in advance dor your help!
Heinz Peter

**Comment by sfmskywalker** [comment]
Hi Heinz Peter,

One easy way to be notified when the workflow executes an activity is to handle the `ActivityExecuting` and/or `ActivityExecuted` notifications.

The former is published when an activity is about to execute and the latter is published after the activity has executed.
They both carry a payload that gives you information about the activity in question.

Let me know if this helps or if you were looking for a different solution.

**Reply by hpposch** [reply]
Hi @sfmskywalker,

thank you for your fast reply!
I had a look inside Elsa.Handlers and created a custom handler.
Afterwards i registered it like that: services.AddNotificationHandlersFrom<ActivityExecutingHandler>();
Everything works as expected!
Thank you very much :)
